### PR TITLE
[ContactStructuralMechanicsApplication] Adding simplified semi-smooth method as alternative for contact

### PR DESCRIPTION
**Pull request by loumalouomega**
In this case the active-inactive set is check just during the first NL iteration

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h [hunk 1]
```diff
@@ -166,67 +166,72 @@ class ALMFrictionalMortarConvergenceCriteria
...
 
-        NodesArrayType& nodes_array = rModelPart.GetSubModelPart("Contact").Nodes();
-
-        #pragma omp parallel for reduction(+:is_converged_active, is_converged_slip)
-        for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i) {
-            auto it_node = nodes_array.begin() + i;
-
-            const double epsilon = it_node->Has(INITIAL_PENALTY) ? it_node->GetValue(INITIAL_PENALTY) : common_epsilon;
+        // We check the active/inactive set during the first non-linear iteration or for the general semi-smooth case
+        if (rModelPart.Is(INTERACTION) || r_process_info[NL_ITERATION_NUMBER] == 1) {
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h:173]: Using `INTERACTION` is misleading imo, why not use another flag? e.g. `SEMI_SMOOTH`?

**Review by philbucher — COMMENTED** [review]
minor, afterwards I would approve

File: applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h (hunk 1 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h:173]: I am not using that flag and the number of flags is limited, I can not ask for a new one every time.

Comment (philbucher) [reply applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h:173]: ok
But you could create your own flags, [like I do](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MappingApplication/custom_utilities/mapper_flags.h) ...
Just saying

**Review by philbucher — APPROVED** [review]

File: applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h (hunk 1 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h:173]: That's a local flag, I cannot set a local flag on a model part (which contains the common flags, I think)

**Pull request merged**